tt_metal/impl/dispatch/command_queue.cpp

```diff
@@ -1223,14 +1237,137 @@ namespace detail {
 void EnqueueRestart(CommandQueue& cq) {
     ZoneScoped;
     detail::DispatchStateCheck(true);
-    // std::shared_future<void> f;
-    // auto& t = cq.submit( [device = cq.device(), cq_id = cq.id()] {
-        cq.device()->hw_command_queue(cq.id()).restart();
-    // }, f);
-    // f.get();
-    // t.reset();
+    cq.device()->hw_command_queue(cq.id()).restart();
+}
+
+}
+
+CommandQueue::CommandQueue(Device* device, uint32_t id) : device_ptr(device), cq_id(id) {
+    bool value = tt::parse_env("TT_METAL_ASYNC_QUEUES", false);
```

Comment:
Can we not just make this an argument to the command queue constructor? Not sure why we need the environment variable. I'm guessing the reasoning was because we want some flexibility at the ttnn level to jump between modes? If so, I would just ping @arakhmati how to cleanly expose this parameter.

Reply:
For bringup only, we should use set_mode() instead and really when async is tested by default we should be in async by default, so then even set_mode() should not be needed except for debugging

Reply:
If this is to make switching between modes easier at the CLI level, I get it, but maybe we should put DEBUG somewhere in the name so people know that this is not the "production" way of turning on this setting. Additionally, maybe replace the variable value with "async_mode"

Reply:
Filed an issue to remove the envvar https://github.com/tenstorrent-metal/tt-metal/issues/5419

FYI @tt-asaigal I also renamed it from TT_METAL_ASYNC_QUEUES to TT_METAL_CQ_ASYNC_MODE, FYI